beacon_node/http_api/src/lib.rs

```diff
@@ -0,0 +1,1693 @@
+
+                        // Use either the supplied slot or all slots in the epoch.
+                        let slots = query.slot.map(|slot| vec![slot]).unwrap_or_else(|| {
+                            epoch.slot_iter(T::EthSpec::slots_per_epoch()).collect()
+                        });
+
+                        // Use either the supplied committee index or all available indices.
+                        let indices = query.index.map(|index| vec![index]).unwrap_or_else(|| {
+                            (0..committee_cache.committees_per_slot()).collect()
+                        });
+
+                        let mut response = Vec::with_capacity(slots.len() * indices.len());
+
+                        for slot in slots {
+                            // It is not acceptable to query with a slot that is not within the
+                            // specified epoch.
+                            if slot.epoch(T::EthSpec::slots_per_epoch()) != epoch {
+                                return Err(warp_utils::reject::custom_bad_request(format!(
+                                    "{} is not in epoch {}",
+                                    slot, epoch
+                                )));
+                            }
+
+                            for &index in &indices {
+                                let committee = committee_cache
+                                    .get_beacon_committee(slot, index)
+                                    .ok_or_else(|| {
+                                    warp_utils::reject::custom_bad_request(format!(
+                                        "committee index {} does not exist in epoch {}",
+                                        index, epoch
+                                    ))
+                                })?;
+
+                                response.push(api_types::CommitteeData {
+                                    index,
+                                    slot,
+                                    validators: committee
+                                        .committee
+                                        .iter()
+                                        .map(|i| *i as u64)
+                                        .collect(),
+                                });
+                            }
+                        }
+
+                        Ok(api_types::GenericResponse::from(response))
+                    })
+                })
+            },
+        );
+
+    // GET beacon/headers
+    //
+    // Note: this endpoint only returns information about blocks in the canonical chain. Given that
+    // there's a `canonical` flag on the response, I assume it should also return non-canonical
+    // things. Returning non-canonical things is hard for us since we don't already have a
+    // mechanism for arbitrary forwards block iteration, we only support iterating forwards along
+    // the canonical chain.
```

Comment:
We can iterate forwards to a single arbitrary head, but that functionality isn't exposed by the `BeaconChain` at the moment. It looks like this API might need forwards iteration forwards across _all_ branches/heads, in which case we're out of luck?

Reply:
I've added a note to #1434 so we can get to this as we finish aligning with the spec.